private/buf/buffetch/proto_file_ref.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020-2021 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buffetch
+
+import (
+	"path/filepath"
+
+	"github.com/bufbuild/buf/private/buf/buffetch/internal"
+	"github.com/bufbuild/buf/private/pkg/normalpath"
+)
+
+var _ ProtoFileRef = &protoFileRef{}
+
+type protoFileRef struct {
+	protoFileRef internal.ProtoFileRef
+}
+
+func newProtoFileRef(internalProtoFileRef internal.ProtoFileRef) *protoFileRef {
+	return &protoFileRef{
+		protoFileRef: internalProtoFileRef,
+	}
+}
+
+func (r *protoFileRef) PathForExternalPath(externalPath string) (string, error) {
```

Comment:
We should make it clear that `PathForExternalPath` can only return one of two values - `".", nil` in the successful case, or `nil, <$some_error>` in the error case. In short, the `externalPath` must be equivalent to the `protoFileRef` - otherwise we won't be able to create a valid relative path from it. 

We could even adapt the implementation so that it's impossible for an `externalPath` to yield anything else (which _should_ already be impossible with how this is used). Again, either the `externalPath is equal to `protoFileRef` and `".", nil` should be returned, or we should return an error.

 For example, if the `protoFileRef` is equal to `foo/bar/baz.proto` and the `externalPath is `foo/bar/baz.proto/qux.proto`, this function will still work as-is since it would treat `baz.proto` as a directory in the latter example.